crypto/x509/t_x509.c

```diff
@@ -94,11 +94,11 @@ int X509_print_ex(BIO *bp, X509 *x, unsigned long nmflags,
         }
         if (l != -1) {
             if (bs->type == V_ASN1_NEG_INTEGER) {
-                l = -l;
+                l = -(unsigned long)l;
```

Comment:
> Casting a signed to unsigned is actually well defined in the C standard

Where? If you refer to "6.3.1.3 Signed and unsigned integers" in "6.3 Conversions", then it it doesn't really remove ambiguity [in my mind], because it discusses rather "narrowing" than "widening" such as conversion to unsigned of same type. I mean that "until the value is in the range of the new type" thing being true from the start. Anyway I suppose that on ones' complement cast to unsigned would simply yield bit pattern of signed presentation. And provided that operations on unsigned are equivalent to those on two's complement 0-(unsigned)-1 would result in 2 on ones' complement system. No?